src/Sylius/Bundle/ShippingBundle/Assigner/ShippingDateAssigner.php

```diff
@@ -14,19 +14,36 @@
 namespace Sylius\Bundle\ShippingBundle\Assigner;
 
 use Sylius\Bundle\ShippingBundle\Provider\DateTimeProvider;
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
 use Sylius\Component\Shipping\Model\ShipmentInterface;
 
 final class ShippingDateAssigner implements ShippingDateAssignerInterface
 {
-    private DateTimeProvider $calendar;
-
-    public function __construct(DateTimeProvider $calendar)
+    public function __construct(private ?DateTimeProvider $calendar, private ?DateTimeProviderInterface $dateTimeProvider = null)
```

Comment:
As we want to replace this argument at the end, I would suggest removal of types to allow both of services be injected into constructor
```suggestion
    public function __construct(private $calendar)
```
or use PHP8.0 union type feature 🎉 
```suggestion
    public function __construct(private DateTimeProvider|DateTimeProviderInterface $calendar)
```

And trigger error if `DateTimeProvider` is provided. If so, it will be assigned to the same field, so the method selection logic needs to be moved to method call. Would be perfect, to do it in private method, as it is required in two places